selfservice/flow/logout/handler.go

```diff
@@ -198,6 +202,16 @@ func (h *Handler) performNativeLogout(w http.ResponseWriter, r *http.Request, _
 		h.d.Writer().WriteError(w, r, err)
 		return
 	}
+	sess, err := h.d.SessionPersister().GetSessionByToken(r.Context(), p.SessionToken, session.ExpandNothing, identity.ExpandNothing)
+	if err != nil {
+		if errors.Is(err, sqlcon.ErrNoRows) {
+			h.d.Writer().WriteError(w, r, errors.WithStack(herodot.ErrForbidden.WithReason("The provided Ory Session Token could not be found, is invalid, or otherwise malformed.")))
+			return
+		}
+
+		h.d.Writer().WriteError(w, r, err)
+		return
+	}
```

Comment:
Not 100% sure it's a good idea to load the session just so we can attach some metadata to the event (Line 226). Is there maybe a less invasive way of getting this data here?

Reply:
All databases, except mysql ofc, support returning deleted rows. You could make the delete call return the session and in the case of mysql do a select before the delete.
Who expects performance when using mysql anyway? :sweat_smile:

Reply:
I'd rather skip the session ID completely. It's not good for much after the session's been revoked anyway. If we find we're missing it in analysis later, we can add it then. But probably YAGNI.

Reply:
I agree with the session ID, but the IdentityID could be interesting

Reply:
`ExpandNothing` on `GetSessionByToken` should be fast enough since it's a global, I think this is fine :) And for logout we should know the user id that is affected!

Reply:
👍